contrib/hamilton/contrib/user/zilto/nixtla_mlforecast/__init__.py

```diff
@@ -0,0 +1,246 @@
+import logging
+from typing import Any, Callable, Iterable, Optional, Union
+
+logger = logging.getLogger(__name__)
+
+from hamilton import contrib
+
+with contrib.catch_import_errors(__name__, __file__, logger):
+    import matplotlib
+    import pandas as pd
+    from mlforecast import MLForecast
+    from mlforecast.target_transforms import BaseTargetTransform
+    from numba import njit
+    from sklearn.base import BaseEstimator
+    from sklearn.ensemble import HistGradientBoostingRegressor, RandomForestRegressor
+    from utilsforecast.evaluation import evaluate
+    from utilsforecast.losses import mse
+    from utilsforecast.plotting import plot_series
+    from window_ops.expanding import expanding_mean
+    from window_ops.rolling import rolling_mean
+
+
+# sklearn compliant models (including XGBoost and LightGBM) subclass BaseEstimator
+MODELS_TYPE = Union[
+    BaseEstimator, list[BaseEstimator], dict[str, BaseEstimator]
+]  # equivalent to mlforecast.core.Models
+LAG_TRANSFORMS_TYPE = dict[int, list[Union[Callable, tuple[Callable, Any]]]]
+DATE_FEATURES_TYPE = Iterable[Union[str, Callable]]
+CONFIDENCE_INTERVAL_TYPE = Optional[list[Union[int, float]]]
+
+
+def base_models() -> MODELS_TYPE:
+    """ML models to fit and evaluate
+    Override with your own models at the Hamilton Driver level.
+    """
+    return [
+        RandomForestRegressor(),
+        HistGradientBoostingRegressor(),
+    ]
+
+
+@njit
+def _rolling_mean_48(x) -> Callable:
+    """Example lag transform defined in a function"""
+    return rolling_mean(x, window_size=48)
+
+
+def lag_transforms() -> LAG_TRANSFORMS_TYPE:
+    """Use operations from `window_ops` (packaged with mlforecast)"""
+    return {
+        1: [expanding_mean],  # function without arguments
+        24: [(rolling_mean, 24)],  # function with arguments as tuple
+        48: [_rolling_mean_48],  # wrapped function
```

Comment:
Why is this warpped? Versus the other one?

Reply:
Because `_rolling_mean_48` is actually a function returning another function. Clarified the comment in the most recent commit.